src/include/duckdb/main/connection.hpp

```diff
@@ -102,7 +103,20 @@ class Connection {
 	void Commit();
 	void Rollback();
 
+	template<typename TR, typename... Args>
+	void CreateFunction(string name, TR (*udf_func)(Args...)) {
+		udf_wrapper.CreateFunction<TR, Args...>(name, udf_func);
+	}
+
+	template<typename TR, typename... Args>
+	void CreateFunction(string name, vector<SQLType> args, SQLType ret_type, TR (*udf_func)(Args...)) {
+		udf_wrapper.CreateFunction<TR, Args...>(name, args, ret_type, udf_func);
+	}
+
 private:
+	//! UDFWrapper to create temporary udf functions with this connection
+	UDFWrapper udf_wrapper;
```

Comment:
Is there a reason that this is an object? I would think all of the functions of UDFWrapper could be implemented as static functions, no?

Reply:
Yeah. Actually, I'd thought to use only static functions, but the ClientContext must be passed as argument for every function such as CreateFunction(..., context), CreateUnary(...,context), CreateBinary(...,context), CreateTernary(...,context) and so on, until the RegisterFunction(..., context) that will use effectively the context to register the function into the catalog. With the context as member class I avoid such an annoying situation.

Another approach to use static functions is the CreateFunction() returns a scalar_function_t and then the connection call the RegisterFunction(name, scalar_function_t, context):

template<typename TR, typename... Args>
void CreateFunction(string name, TR (*udf_func)(Args...)) {
	scalar_fucntion_t function = udf_wrapper.CreateFunction<TR, Args...>(name, udf_func);
        udf_wrapper.RegisterFunction<TR, Args...>(name, function, context);
}

I think both approaches are just different way of implementation, but if you decide for static functions I can change the code, no worries.